x-pack/legacy/plugins/canvas/public/applications/canvas.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { Store } from 'redux';
+import ReactDOM from 'react-dom';
+import { I18nProvider } from '@kbn/i18n/react';
+import { Provider } from 'react-redux';
+
+import { AppMountParameters, CoreStart } from 'kibana/public';
+
+// @ts-ignore Untyped local
+import { App } from '../components/app';
+import { KibanaContextProvider } from '../../../../../../src/plugins/kibana_react/public';
+
+export const renderApp = (
+  coreStart: CoreStart,
+  plugins: object,
+  { element }: AppMountParameters,
+  canvasStore: Store
+) => {
+  ReactDOM.render(
+    <KibanaContextProvider services={{ ...coreStart, ...plugins }}>
```

Comment:
nit: The `KibanaContextProvider` uses react context to provide enable deeply nested react components to directly consume these services. I didn't look closely at the Canvas code, but since you seem to be using Redux, you probably won't need this context and would instead consume Core API's from reducers.

Reply:
If you're using the hooks from `kibana-react` then this is correct.